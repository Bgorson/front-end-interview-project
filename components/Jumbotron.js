import React from 'react';
import Image from 'next/image';
import styles from './Jumbotron.module.css';
export default function Jumbotron({aisleTitle}) {

    return (
        <div className={styles.jumboTron}>
            <span className={styles.jumboImage}>
                <Image src="/background.png" alt="wine" height="200" width="300"/>
            </span>
            <div className={styles.jumboText}>
                <h1>{aisleTitle} Aisle</h1>
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temre et dolierunt mollit anim id est laborum.</p>
            </div>
        </div>
    );
}