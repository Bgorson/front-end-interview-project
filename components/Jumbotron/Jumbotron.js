import Image from 'next/image';
import styled from 'styled-components';
import styles from './Jumbotron.module.css';

const StyledJumbotron =styled.div `{
    display:flex;
    justify-content:space-between;
    flex-wrap:no-wrap;
    margin-top:1rem;
    margin-bottom:1rem;
    @media (max-width: 600px) {
          display:block;
          width:100%;
          margin-top:0;
        }
}`;

const StyledJumboImage =styled.span `{
    max-width: 100%;
    height: auto;
    margin-right:5rem;
    @media (max-width: 600px) {
            display: block;
            margin:0;
            width:100%;
        }
}`;

const StyledJumboText =styled.div `{
    flex:1;
    margin:auto;
    width:90%;
}`;

export default function Jumbotron({aisleTitle}) {

    return (
        <StyledJumbotron>
            <StyledJumboImage>
                <Image className= {styles.mainImage} src="/background.png" alt="wine" height="200" width="400"/>
            </StyledJumboImage>
            <StyledJumboText>
                <h1>{aisleTitle} Aisle</h1>
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temre et dolierunt mollit anim id est laborum.</p>
            </StyledJumboText>
        </StyledJumbotron>
    );
}