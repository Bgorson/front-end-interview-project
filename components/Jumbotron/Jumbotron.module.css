.jumboTron{
    display:flex;
    justify-content:space-between;
    flex-wrap:no-wrap;
    margin-top:1rem;
    margin-bottom:1rem;
}
.jumboText{
    flex:1;
}
.jumboImage{
    max-width: 100%;
    height: auto;
    margin-right:5rem;


}
@media (max-width: 600px) {
    .jumboTron{
      display:block;
      width:100%;
    }
    .jumboImage{
        display: block;
        margin:0;
        width:100%;
    }
  }