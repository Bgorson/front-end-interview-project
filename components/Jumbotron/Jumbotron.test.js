import renderer from 'react-test-renderer';
import Jumbotron from './Jumbotron';

it('should render correctly', () => {
    const component = renderer.create(<Jumbotron/>).toJSON();

    expect(component).toMatchSnapshot();
});