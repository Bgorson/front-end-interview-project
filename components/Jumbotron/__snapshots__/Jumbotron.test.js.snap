// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="Jumbotron__StyledJumbotron-sc-1k7fk82-0 fMdatd"
>
  <span
    className="Jumbotron__StyledJumboImage-sc-1k7fk82-1 emKWjm"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;charset=utf-8,<svg width=\\"300\\" height=\\"200\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
          style={
            Object {
              "display": "block",
              "maxWidth": "100%",
            }
          }
        />
      </div>
      <img
        alt="wine"
        className="mainImage"
        decoding="async"
        src="/_next/image?url=%2Fbackground.png&w=1024&q=75"
        srcSet="/_next/image?url=%2Fbackground.png&w=320&q=75 1x, /_next/image?url=%2Fbackground.png&w=768&q=75 2x, /_next/image?url=%2Fbackground.png&w=1024&q=75 3x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
  </span>
  <div
    className="Jumbotron__StyledJumboText-sc-1k7fk82-2 dCeRZI"
  >
    <h1>
       Aisle
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temre et dolierunt mollit anim id est laborum.
    </p>
  </div>
</div>
`;
